admin-dev/themes/default/js/bundle/product/form.js

```diff
@@ -961,6 +961,8 @@ var form = (function() {
   }
 
   function switchLanguage(iso_code) {
+    $('#form_switch_language option[value!="' + iso_code + '"]').attr('selected', false);
+    $('#form_switch_language option[value="' + iso_code + '"]').attr('selected', true);
```

Comment:
You should be able to control it using `.val()` ([see example](https://select2.github.io/examples.html#programmatic))

Maybe the problem is that the `<option>`'s `data-idlang` attributes seem to have an extra `"`